web/ui/src/features/component/ComponentBody.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { Fragment } from 'react';
+import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
+
+import { riverStringify } from '../river-js/stringify';
+import Table from '../widgets/Table';
+
+import { style } from './style';
+import { PartitionedBody } from './types';
+
+import styles from './ComponentView.module.css';
+
+interface ComponentBodyProps {
+  partition: PartitionedBody;
+}
+
+const TABLEHEADERS = ['Name', 'Value'];
+
+const ComponentBody = ({ partition }: ComponentBodyProps) => {
+  const sectionClass = partition.key.length === 1 ? '' : styles.nested;
+
+  const renderTableData = () => {
+    return partition.attrs.map(({ name, value }, index) => {
+      return (
+        <tr key={name}>
+          <td className={styles.nameColumn}>{name}</td>
+          <td>
+            <pre className={styles.pre}>
+              <SyntaxHighlighter language="javascript" style={style}>
+                {riverStringify(value)}
+              </SyntaxHighlighter>
+            </pre>
+          </td>
+        </tr>
+      );
+    });
+  };
+
+  return (
+    <>
+      <section id={partition.key.join('-')} className={sectionClass}>
+        {
+          // If the partition only has 1 key, then make it an h2.
+          // Otherwise, make it an h3.
+          partition.displayName.length === 1 ? (
+            <h2>{partition.displayName}</h2>
+          ) : (
+            <h3>
+              {partition.displayName.map((val, idx) => {
+                return (
+                  <Fragment key={idx.toString()}>
```

Comment:
using an index as the key is not recommended practice. https://react.dev/learn/rendering-lists#why-does-react-need-keys See the "Pitfall" section.